apps/api/src/service/authService.ts

```diff
@@ -76,3 +82,46 @@ export async function isAssociatedRep(session: Partial<SessionData>, application
 
 	return false;
 }
+
+/**
+ * Handler to enforce Request is Authorized
+ * Insures Type safety for downstream Request Handler due to limitations with Express Type definitions
+ * @param request Express request
+ * @returns boolean
+ */
+export const isAuthenticatedRequest = (request: Request): request is AuthorizedRequest => {
+	return isRequestWithSession(request);
+};
+
+/**
+ * Validate User is allowed access to this specific Application based on userRole or userId
+ * @param request AuthorizedRequest with session, user and userId
+ * @returns boolean
+ */
+export async function canAccessRequest(
+	request: AuthorizedRequest,
+): AsyncResult<AuthorizedRequest, 'FORBIDDEN' | 'NOT_FOUND' | 'SYSTEM_ERROR'> {
```

Comment:
Could we refactor this to not take the `request` object, but instead take the `session` and `applicationId` as parameters?